event hubs migration guide.md

```diff
@@ -0,0 +1,206 @@
+
+For example, this code which receives from a partition in V2:
+
+```typescript
+const client = EventHubClient.createFromConnectionString(connectionString);
+const rcvHandler = client.receive(partitionId, onMessageHandler, onErrorHandler, {
+  eventPosition: EventPosition.fromStart(),
+  consumerGroup: consumerGroupName
+});
+await rcvHandler.stop();
+```
+
+Becomes this in V5:
+
+```typescript
+const eventHubConsumerClient = new EventHubConsumerClient(consumerGroupName, connectionString);
+
+const subscription = eventHubConsumerClient.subscribe(
+  partitionId, {
+    processInitialize: (initContext) => {
+      initContext.setStartingPosition(EventPosition.fromStart());
+    },
+    processEvents: onMessageHandler,
+    processError: onErrorHandler
+});
+  
+await subscription.close();
+```
+
+See [`receiveEvents.ts`](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) 
+for a sample program demonstrating this.
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient` for sending events
+
+In V2, there were multiple options on how to send data.
+
+In V5, this has been consolidated into a more efficient `sendBatch` method. 
+Batching merges information from multiple messages into a single send, reducing
+the amount of network communication needed vs sending messages one at a time.
+
+So in V2:
+```typescript
+const eventsToSend = [
+  // events go here
+];
+
+const client = EventHubClient.createFromConnectionString(connectionString);
+console.log(`Sending event: ${eventData.body}`);
+
+// Would fail if the total size of events exceed the max size supported by the library.
+await client.send(eventsToSend, partitionId);
+```
+
+In V5:
+```typescript
+const producer = new EventHubProducerClient(connectionString);
+
+const eventsToSend = [
+  // events go here
+];
+
+let batch = await producer.createBatch();
+
+for (const event of eventsToSend) {
+  // messages can fail to be added to the batch if they exceed the maximum size configured for
+  // the EventHub.
+  const isAdded = batch.tryAdd({ body: event });
+  
+  if (!isAdded) {
+    if (batch.count === 0) {
+      // if we can't add it and the batch is empty that means the message we're trying to send
+      // is too large, even when it would be the _only_ message in the batch.
+      //
+      // To fix this you'll need to split the message up across multiple batches or 
+      // skip it. In this example, we'll skip the message.
+      console.log(`Message was too large and can't be sent until it's made smaller. Skipping...`);
+      continue;
+    }
+
+    // otherwise this just signals a good spot to send our batch
+    console.log(`Batch is full - sending ${batch.count} messages as a single batch.`)
+    await producer.sendBatch(batch);
+
+    // and create a new one to house the next set of messages
+    batch = await producer.createBatch();
+    continue;
```

Comment:
In this model, wont the last event that resulted in `isAdded` being `false` getting skipped?

Reply:
Only with a binary-based computer.

(fixed. Also, fixed the sample it was based on)